activerecord/lib/active_record/relation/batches.rb

```diff
@@ -288,32 +324,39 @@ def apply_limits(relation, start, finish, order)
       end
 
       def apply_start_limit(relation, start, order)
```

Comment:
Not critical but perhaps we could expect this method to receive a `Hash` of column=>order so we don't need to call `get_the_order_of_primary_key` in both methods. In this case `apply_limits` will have to change to something like:
```ruby
      def apply_limits(relation, start, finish, order)
        orders = get_the_order_of_primary_key(order) if start || finish
        relation = apply_start_limit(relation, start, orders) if start
        relation = apply_finish_limit(relation, finish, orders) if finish
        relation
      end
```